test/metabase/driver/sql/query_processor_test.clj

```diff
@@ -13,6 +13,21 @@
             [metabase.util.honeysql-extensions :as hx]
             [schema.core :as s]))
 
+(deftest sql-source-query-validation-test
+  (testing "[[sql.qp/sql-source-query]] should throw Exceptions if you pass in invalid nonsense"
+    (doseq [params [nil [1000]]]
+      (testing (format "Params = %s" (pr-str params))
+        (is (instance? metabase.driver.sql.query_processor.SQLSourceQuery
+                       (sql.qp/sql-source-query "SELECT *" params)))))
+    (is (thrown-with-msg?
+         clojure.lang.ExceptionInfo
+         #"Expected native source query to be a string, got: clojure.lang.PersistentArrayMap"
+         (sql.qp/sql-source-query {:native "SELECT *"} nil)))
+    (is (thrown-with-msg?
+         clojure.lang.ExceptionInfo
+         #"Expected native source query parameters to be sequential, got: java.lang.Long"
+         (sql.qp/sql-source-query "SELECT *" 1000)))))
```

Comment:
This is a test for the aforementioned better-error-message stuff I also added here